packages/neon/neon/lib/src/utils/global_options.dart

```diff
@@ -91,26 +91,29 @@ class GlobalOptions extends OptionsCollection {
   }
 
   void updateAccounts(final List<Account> accounts) {
-    initialAccount.values = {
-      for (final account in accounts) account.id: (final context) => account.humanReadableID,
-    };
+    initialAccount.values = Map.fromEntries(
+      accounts.map(
+        (final account) => MapEntry(account.id, (final context) => account.humanReadableID),
+      ),
+    );
 
-    if (accounts.tryFind(initialAccount.value) == null) {
+    if (!initialAccount.values.containsKey(initialAccount.value)) {
       initialAccount.reset();
     }
```

Comment:
I feel like this check should be done automatically if the value or the values are updated (you repeated the same below). That will also solve https://github.com/nextcloud/neon/issues/892

Reply:
I'll also do this as a followup.
I think I already tried this when doing the change a month ago but can't really remember why it failed.